src/main.rs

```diff
@@ -1,3 +1,132 @@
+#[macro_use]
+extern crate clap;
+
+mod options;
+use crate::options::CLIOptions;
+use clap::App;
+use std::{fs, fs::File, io, path::Path};
+
 fn main() {
-    println!("Hello, world!!!");
+    let yaml = load_yaml!("../cli_def/en_us.yml");
+    let matches = App::from_yaml(yaml).get_matches();
+
+    let raw_options = CLIOptions::from_clap_matches(&matches);
+
+    let options = raw_options.process_options();
+
+    for filename in options.files {
+        let file = File::open(filename).unwrap();
+        unzip_file(
+            file,
+            &std::path::PathBuf::from(&raw_options.destination_folder),
+        );
+    }
+}
+
+fn unzip_file(file: File, directory: &Path) {
+    let mut archive = zip::ZipArchive::new(file).unwrap();
+
+    let files_indexes = 0..archive.len();
+
+    files_indexes
```

Comment:
We might want to map this to all the Files in here, instead of passing the index and in `unzip_archive` we can work with `ZipFile` instead. There is also a way to collect all the Results in a iterator and depending of what we want with the error, fail the whole process or just collect them so we can show later which files were broken.

Reply:
That is what I though, but then I readed the official documentation and seems that you can only iterate indexes. https://docs.rs/zip/0.5.3/zip/read/struct.ZipArchive.html

What is the method you are speaking about?

Reply:
Not speaking about method, just to do:
`(0..archive.len().map(|file_index| {
   let file = archive.by_index(i).unwrap();
   unzip_archive(file, &mut archive, directory)
});`

I.e. pass the file, not an index.

Reply:
Ahh yeah, and that way you don't need a `&mut archive` as the `by_index()` method needs a mutable reference to the Archive (`pub fn by_index<'a>(&'a mut self, file_number: usize)`)